internal/stopwatch/stopwatch.go

```diff
@@ -0,0 +1,175 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package stopwatch
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// Stopwatch is a simple interface to measure time.
+// Stopwatch implementations are not thread-safe.
+type Stopwatch interface {
```

Comment:
I **strongly** recommend sorting `Stopwatch` methods by "type of use":
- Control whether it's counting or not:
  - `Start`
  - `Stop`
- Inspect resource consumption
  - `Remaining` 
  - `Spent`
  - `Expired`